astroquery/esasky/core.py

```diff
@@ -465,13 +479,19 @@ def _sanitize_input_table_list(self, table_list):
         if (isinstance(table_list, commons.TableList)):
             return table_list
         raise ValueError("Query_table_list must be an astropy.utils.TableList")
+    
+    def _sanitize_input_row_limit(self, row_limit):
+        if (isinstance(row_limit, int) or row_limit == self.__DEFAULT_SOURCE_LIMIT):
+            return row_limit
+        
+        raise ValueError("Row_limit must be an integer")
       
-    def _get_maps_for_mission(self, maps_table, mission, download_directory, cache):
+    def _get_maps_for_mission(self, maps_table, mission, download_dir, cache):
```

Comment:
This was already corrected in master.  You should rebase against (or merge) master.